src/figuresClasses.ts

```diff
@@ -1,19 +1,70 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
 export class Triangle {
+  shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle sides');
+    }
+  }
 
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
```

Comment:
make sense to desctruct `this` as it will make a code shorter and cleaner 
`const { a, b, c } = this;`